base/compiler/ssair/inlining.jl

```diff
@@ -1015,12 +1038,22 @@ function assemble_inline_todo!(ir::IRCode, sv::OptimizationState)
 
         nu = countunionsplit(sig.atypes)
         if nu == 1 || nu > sv.params.MAX_UNION_SPLITTING
+            if !isa(info, MethodMatchInfo)
+                info = nothing
+            end
+            infos = [info]
             splits = Any[sig.atype]
         else
             splits = Any[]
             for union_sig in UnionSplitSignature(sig.atypes)
                 push!(splits, argtypes_to_type(union_sig))
             end
+            if !isa(info, UnionSplitInfo)
+                infos = fill(nothing, length(splits))
```

Comment:
wrong object type?

Reply:
You mean it should be Vector{Any}? Yeah, let me do that.